feat: add NcFormGroupButton - an interactive button/link element of NcFormGroup

ShGKme opened the pull request:
### ☑️ Resolves

- Fix: https://github.com/nextcloud-libraries/nextcloud-vue/issues/7123
- An interactive button-like element of a form. Can be used for consistent form items in a stack.
- ⚠️ Current questions: Component name?

### 🖼️ Screenshots

<img width="817" height="325" alt="image" src="https://github.com/user-attachments/assets/a714df58-3570-4384-ade3-15f674604a37" />

### 🏁 Checklist

- [ ] ⛑️ Tests are included or are not applicable
- [x] 📘 Component documentation has been extended, updated or is not applicable
- [x] 2️⃣ Backport to `stable8` for maintained Vue 2 version or not applicable

ShGKme commented on src/components/NcFormItem/NcFormItem.vue:
```diff
@@ -0,0 +1,242 @@
+	 * RouterLink's to props and a trigger to use RouterLink.
+	 * Note: This takes precedence over the href attribute.
+	 */
+	to?: RouteLocationRaw
```
You by complexity you mean the `useButtonLink` - I'd also use it for `NcListItem` (instead of `<a href="#">` and `NcAppnavigationItem`.

I don't have a case for a router link in mind, but for me it seems just a link case in general in an SPA.

Like if you have extended settings in another route. Or a link to the "Note to self".

Otherwise people will use a button with `@click="router.push"`.

susnux replied:
The only thing I do not really like is the name :see_no_evil: 
It is not a form element, it is neither an input nor a button, its an link or list element.

It is not even used in any form, its first usage will be in the app settings, so it would be a settings item.
Or something more generic, but not really any kind of form related.

> I'd also use it for NcListItem

It would be really nice to have it even without any interactive wrapper - but unrelated here.

ShGKme replied:
However, I agree that the current name is not perfect.

But I didn't come up with a better naming :(

ShGKme replied:
Is there anything that would prevent using new design in any form?

Fieldset, radio groups. All the new components are very classic parts of a form.

Maybe except this one item specifically.

ShGKme commented:
- Adjusted according to the design review:
  - Icon size: 20px
  - Added border like on other buttons for Nextcloud 32
  - Adjusted paddings
  - Adjusted colours, but with new colours...
- Instead of `description` new props for optional text are `supline` (above the text) and `subline` (below)

ShGKme commented:
Renamed to `NcFormGroupButton`, where `NcFormGroup` is a visual group of items.
Yet not the final name 👀

ShGKme commented:
- Updated border color according to the design review

kra-mo commented:
> ⚠️ Current questions: Component name?

Maybe something like `NcActionField`, or if we would prefer more specific ones later like copy, then `NcBaseField` so that we could have an `NcCopyField` or something?

But in general, I would prefer if we could call all of these form components fields.

kra-mo commented:
> This confuses with NcActions and all its NcActionX sub-components.

Sure, I did think of that. But at the end of the day, action is such a generic term that I'm not sure anything else could cover all of this. Maybe it's fine to expand our use of the word as well?

But if we would prefer more specific descendants, then base would probably be better anyway.

> But its not a field like a text input its currently only a copy button or an external link.

Field in the context of forms just means an element. At least that's how I view it, and I think it would be a nice way to tie it all up. "A form has fields and the fields are grouped."

> Vue naming rules

Well alright, although if we do establish the convention that you use fields in forms, I feel like that would be clear enough and we would avoid long component names like `NcFormGroupCopyButton` in favor of `NcCopyField`.

You have the final say, but at least internally as designers, I think we'll continue to call it fields so it would be nice to have consistency. At least from a design POV, this isn't really a "button", just a field/form item/row with information and an associated action.
